perception/autoware_detected_object_validation/src/lanelet_filter/lanelet_filter.cpp

```diff
@@ -73,6 +76,37 @@ ObjectLaneletFilterNode::ObjectLaneletFilterNode(const rclcpp::NodeOptions & nod
     std::make_unique<autoware::universe_utils::DebugPublisher>(this, "object_lanelet_filter");
   published_time_publisher_ =
     std::make_unique<autoware::universe_utils::PublishedTimePublisher>(this);
+  viz_pub_ =
+    this->create_publisher<visualization_msgs::msg::MarkerArray>("~/debug/marker", rclcpp::QoS{1});
+}
```

Comment:
It might be better to create the publisher when the `filter_settings_.debug == true` since it only be used in `publishDebugMarkers` that is also only get called when `filter_settings_.debug == true`.

Reply:
Added code that creates publisher only whether there is `filter_settings_.debug == true`.